src/main/java/com/google/jenkins/plugins/k8sengine/Manifests.java

```diff
@@ -63,16 +76,71 @@ public String getKind() {
     }
 
     /** @return The name. */
+    public Optional<String> getName() {
+      return getMetadata().isPresent()
+          ? Optional.of((String) getMetadata().get().get("name"))
+          : Optional.empty();
+    }
+
+    /**
+     * Ensures this {@link ManifestObject} has labels, modifying in-place as needed, finally
+     * returning the labels.
+     *
+     * @return The labels for this {@link ManifestObject}.
+     */
     @SuppressWarnings("unchecked")
-    public String getName() {
-      Map<String, Object> metadata = (Map<String, Object>) source.get("metadata");
-      return (String) metadata.get("name");
+    public Map<String, String> getOrCreateLabels() {
+      Map<String, Object> metadata = getOrCreateMetadata();
+
+      if (!metadata.containsKey("labels")) {
+        metadata.put("labels", new LinkedHashMap<String, String>());
+      }
+
+      return (Map<String, String>) metadata.get("labels");
+    }
+
+    /**
+     * Adds the specified label key and value to this {@link ManifestObject}'s metadata labels. Will
+     * ensure a label map exists upon execution.
+     *
+     * @param key The key of the label to be added.
+     * @param value The value of the label to be added.
+     */
+    public void addLabel(String key, String value) {
+      Map<String, String> labels = getOrCreateLabels();
+      // Add the specified label ensuring no duplicate values.
+      Set<String> labelValues =
+          labels.get(key) != null
+              ? new HashSet<>(Arrays.asList(labels.get(key).split(",")))
+              : new HashSet<>();
+      labelValues.add(value);
+      labels.put(key, String.join(",", labelValues));
     }
 
     /** @return The description of the object in {ApiVersion}/{Kind}: {Name} */
     public String describe() {
       return String.format("%s/%s: %s", getApiVersion(), getKind(), getName());
```

Comment:
Because getName is now optional, recommend using `getName().get()`, or `getName().orElse("<some default>")`.

Reply:
Good catch. Thanks.